### PR TITLE
Ebnt 126 pytorch

### PR DESCRIPTION
**Pull request by i-a-andrianov**
PyTorch integration: dataset types and model wrapper

### CODE REVIEW

File: tests/ext/torch/test_dataset.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    assert tdt.types[0].shape == (5, 10)
+    assert tdt.types[0].dtype == 'float32'
+    assert tdt.types[0].list_size() == 5
+    assert tdt.types[1].shape == (5, 5)
+    assert tdt.types[1].dtype == 'float32'
+    assert tdt.types[1].list_size() == 5
+
+    tensor_list_deser = tdt.deserialize(tdt.serialize(tensor_list))
+    assert len(tensor_list) == len(tensor_list_deser)
+    assert all(torch.equal(tensor, tensor_deser) for tensor, tensor_deser in zip(tensor_list, tensor_list_deser))
```
Comment (mike0sv) [inline tests/ext/torch/test_dataset.py]: maybe add some tests for .get_spec (call .runtime.openapi.spec.type_to_schema on dataset type)

File: src/ebonite/ext/torch/dataset.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            subtype = self._get_dtype_from_str(self.dtype)
+        else:
+            subtype = self._get_subtype(shape[1:])
+        return SizedTypedListType(shape[0], subtype)
+
+    @staticmethod
+    def _get_dtype_from_str(dtype_str: str):
+        # `torch.float32` and others are singleton instances of type `torch.dtype` and not types
+        # thus we use corresponding built-in types instead
+        known_types = ['float', 'int', 'complex']
```
[the remark this answers is not in the document]
Comment (i-a-andrianov) [reply src/ebonite/ext/torch/dataset.py]: no, eliminated it

File: tests/ext/torch/test_dataset.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import pytest
+
+from ebonite.core.analyzer.dataset import DatasetAnalyzer
+from ebonite.utils.importing import module_importable
+
+
+@pytest.mark.skipif(not module_importable('torch'), reason='PyTorch is not installed')
```
[the remark this answers is not in the document]
Comment (i-a-andrianov) [reply tests/ext/torch/test_dataset.py]: ok

File: tests/ext/torch/test_model.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import pytest
+
+from ebonite.core.analyzer.model import ModelAnalyzer
+from ebonite.utils.importing import module_importable
+
+MyNet = None
+if module_importable('torch'):
```
[the remark this answers is not in the document]
Comment (i-a-andrianov) [reply tests/ext/torch/test_model.py]: ok

File: tests/ext/torch/test_dataset.py (hunk 1 above)
Comment (i-a-andrianov) [reply tests/ext/torch/test_dataset.py]: done

File: src/ebonite/ext/torch/dataset.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    :class:`TorchTensorDatasetType` and :class:`TorchTensorListDatasetType`
+    """
+    def must_process(self, obj) -> bool:
+        return isinstance(obj, torch.Tensor) or \
+               isinstance(obj, list) and all(isinstance(e, torch.Tensor) for e in obj)
+
+    def process(self, obj) -> DatasetType:
+        if isinstance(obj, torch.Tensor):
+            return TorchTensorDatasetType(tuple(obj.shape), str(obj.dtype)[len('torch.'):])
+        return TorchTensorListDatasetType([self.process(e) for e in obj])
```
[the remark this answers is not in the document]
Comment (i-a-andrianov) [reply src/ebonite/ext/torch/dataset.py]: done

File: tests/ext/torch/test_dataset.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    with pytest.raises(ValueError):
+        DatasetAnalyzer.analyze([first_tensor, second_tensor])
+
+
+@pytest.mark.skipif(not module_importable('torch'), reason='PyTorch is not installed')
+def test_torch__single_tensor(first_tensor):
+    import torch
+
+    # this import ensures that this dataset type is registered in `DatasetAnalyzer`
+    from ebonite.ext.torch.dataset import TorchTensorDatasetType  # noqa
```
[the remark this answers is not in the document]
Comment (i-a-andrianov) [reply tests/ext/torch/test_dataset.py]: for model wrapper it works but for dataset type it doesn't: if I eliminate import I get "no hook found"

[the remark this answers is not in the document]
Comment (mike0sv) [reply tests/ext/torch/test_dataset.py]: that's strange, in similar case with xgboost everything is fine. Do you import dataset hook in ext.torch.__init__.py ?

**Pull request merged**